### PR TITLE
Rollback bolt tx on mapper open error

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This commit fixes `SelectMapper.Open()` so that it properly rolls back transactions. Previously, this caused transactions to stay open indefinitely which caused mmap resizes to hang indefinitely.

### CODE REVIEW

File: tsdb/mapper.go [hunk 1]
```diff
@@ -114,185 +112,192 @@ func (lm *SelectMapper) Open() error {
...
-		// For GROUP BY time queries, limit the number of data points returned by the limit and offset
-		d, err := lm.selectStmt.GroupByInterval()
-		if err != nil {
-			return err
-		}
-		lm.intervalSize = d.Nanoseconds()
-		if lm.queryTMin == 0 || lm.intervalSize == 0 {
-			lm.numIntervals = 1
-			lm.intervalSize = lm.queryTMax - lm.queryTMin
+	if err := func() error {
```
Comment (benbjohnson) [inline tsdb/mapper.go]: This line and the conditional block at the bottom are the only changes in this PR.

File: tsdb/mapper.go [hunk 2]
```diff
@@ -114,185 +112,192 @@ func (lm *SelectMapper) Open() error {
...
+		if lm.rawMode && len(lm.selectFields) == 0 {
+			// None of the SELECT fields exist in this data. Wipe out all tagset cursors.
+			lm.cursors = nil
+		}
+
+		return nil
+	}(); err != nil {
+		lm.tx.Rollback()
+		return err
 	}
```
Comment (benbjohnson) [inline tsdb/mapper.go]: Here's the conditional block.

**Comment by otoolep** [comment]
+1

**Comment by jwilder** [comment]
:+1:

**Comment by corylanou** [comment]
+1

**Comment by otoolep** [comment]
Great fix.

**Pull request merged**